### PR TITLE
adding styling and data proptypes

### PR DESCRIPTION
**Pull request by readeral**
as it looks

### CODE REVIEW

File: packages/calendar/calendar.js [hunk 1]
```diff
@@ -217,7 +217,29 @@ export default class Calendar extends React.Component {
...
+                                                margin: 0 2px;
+                                                white-space: nowrap;
+                                                overflow: hidden;
+                                                text-overflow: ellipsis;
+                                                font-size: 14px;
+                                                color: white;
+                                                font-family: 'arial';
+                                              `}
+                                            >
+                                              <span style={{fontWeight: '700'}}>
```
Comment (BarryThePenguin) [inline packages/calendar/calendar.js]: Try and stick with mineral-ui components, like `Text`. Otherwise we lose out on the theme support

```

**Comment by BarryThePenguin** [comment]
Noticed in other parts that there is a lot of css styling other elements, eg

```js
const headingRow = css`
  border-spacing: 4px 0px;
  th {
    height: 1rem;
    padding: 4px 0;
    vertical-align: middle;
    background: lightblue;
    text-align: center;
    font-family: 'helvetica';
  }
  th:first-of-type {
    width: 30px;
  }
`;
```

Try and keep css scoped to a single element. If you want to use sudo selectors, emotion supports parent selectors, eg

```js
const headingRow = css`
  height: 1rem;
  padding: 4px 0;
  vertical-align: middle;
  background: lightblue;
  text-align: center;
  font-family: 'helvetica';

  &:first-of-type {
    width: 30px;
  }
`;
```

**Comment by readeral** [comment]
So the reason I (started) doing this is because of line 143-153 - given css enabled me to target the child th, I did, rather than append the class to every th. It also reduced the number of css declarations I had, as it was already getting rather long.

However, later on I got lazy and instead of appending the class to my mapped items, I took the same approach.

I wonder whether I should split any of this code out - given it's all one table (essentially) it makes sense to me to keep it all together, it just means as much CSS declaration as there is code.

File: packages/calendar/calendar.js [hunk 2]
```diff
@@ -17,28 +18,21 @@ const tableReset = css`
...
-    background: lightblue;
-    text-align: center;
-    font-family: 'helvetica';
-  }
   th:first-of-type {
     width: 30px;
   }
 `;
 
+const headTh = styled('th')`
```
Comment (BarryThePenguin) [inline packages/calendar/calendar.js]: Make sure to use capitals `HeadTh`, it's a react component now

Comment (readeral) [reply packages/calendar/calendar.js]: I've screwed it up with trying to use stash anyway. Will ask your help at 1

**Pull request closed without merging**